stl/inc/xutility

```diff
@@ -2479,17 +2479,31 @@ namespace ranges {
     _NODISCARD constexpr auto _As_const_pointer(const _Ty* _Ptr) noexcept {
         return _Ptr;
     }
+
+    template <class _Ty>
+    using _Begin_adapted = decltype(_RANGES begin(_RANGES _Possibly_const_range(_STD declval<_Ty&>())));
+
+    template <class _Ty>

```

Comment:
I'd prefer a transition comment here (I like the `_Begin_adapted` stuff permanently, though). Same for the rest.